docs/tcpAgent.md

```diff
@@ -0,0 +1,140 @@
+These were formerly known as JNLP agents, but that name was erroneous and confusing.
+This document describes some of the primary mechanisms for launching inbound TCP agents.
+
+For additional information about Jenkins agents see [Distributed builds](https://wiki.jenkins.io/display/JENKINS/Distributed+builds#Distributedbuilds-HavemasterlaunchagentonWindows).
+
+## Launch mechanisms
+Part of the agent status page for TCP agents looks something like this:
+![Tcp agent status UI](tcpAgentStatus.jpg)
+This shows a couple of launch methods, but there are other mechanisms.
+If you use a different mechanism than clicking the "Launch" button, this page contains parameters that you will need. 
+The long string of hex digits is a secret key that the client needs to establish the connection. 
+Depending on how you connect you may also need the agent name, "AgentName" in this example, or the (misnamed) jnlpUrl, "https://myjenkins.example.com/computer/AgentName/slave-agent.jnlp".
+
+### "Launch" button
+This is the only launch mechanism that actually uses JNLP.
+When properly configured and supported, clicking this button causes a JNLP file to be downloaded via the web browser and launched with Java WebStart. 
+
+While this is the simplest mechanism it has many drawbacks.
+This is an outdated technology that is being phased out.
+It requires an interactive user.
+If the agent terminates the user will have to click the button again.
+This mechanism is not recommended, is deprecated, and may not be supported or available.
+
+### Download JNLP file
+Another mechanism, shown in the above status page fragment, runs the agent from a script or command-line to retrieve the JNLP file. 
+This mechanism does not use JNLP or WebStart but uses the downloaded file to obtain connection information. 
+Internally, this is a two-step process: 1) connect to an HTTP(S) port to retrieve the connection information, and 2) extract that information and connect to the TCP port.
+
+Before invoking this command you must download the "agent.jar" file.
+The correct version can be obtained from your Jenkins server at something like "https://myjenkins.example.com/jnlpJars/agent.jar".
+These operations can be combined in a script, cron job, or service.
+
+A typical example of this command, as shown in the above image, is
+``` 
+java -jar agent.jar \
+  -jnlpUrl <jnlp url> \
+  -secret <secret key> \
+  -workDir <work directory> 
+```
+
+### Connect directly to TCP port
+With the latest versions of the Remoting library (since TODO), it is possible to connect directly to the TCP port. 
+This skips the download of the connection information file, described in step 1) above. 
+To use this you must provide some of the information that the downloaded file would contain.
+
+You must provide the agent name. 
+This is easily obtained from the agent status page or other places.
+
+You need to specify the host and port, in a typical `HOST:PORT` fashion. 
```

Comment:
If you do manage to support IPv6 here, it might be good to note that the IPv6 address should be surrounded in square brackets just like it would in a URL.

Reply:
Good point. I'll adjust the implementation.